app/grandchallenge/algorithms/forms.py

```diff
@@ -566,7 +555,11 @@ def __init__(
         phase,
         **kwargs,
     ):
-        super().__init__(*args, user=user, phase=phase, **kwargs)
+        self._user = user
+        self._phase = phase
+
+        super().__init__(*args, **kwargs)
```

Comment:
This construction is a bit strange now. The developer needs to know to set these attributes in order to use `UserAlgorithmsForPhaseMixin`. Isn't it better to keep that as a responsibility of the mixin?

I think this pattern is repeated with having to set `self.additional_inputs`, so I think there is still a design problem.

Reply:
Hm, I'm unsure how to properly pass along the user argument to the mixins. Both `UserAlgorithmsForPhaseMixin` and `AdditionalInputsMixin` need it, but they are not always used together, so if I modified `UserAlgorithmsForPhaseMixin`  to pass on the user argument in the call to `super()`, I run into issues in places where that mixin is used without `AdditionalInputsMixin`.

Reply:
The only way I found to make this work is by passing the extra kwargs via `kwargs` and not declare them explicitly in the mixin's init method. I haven't seen that pattern elsewhere in the codebase though, and I'm wondering if I should then raise an error if a required argument is not present in kwargs.

Reply:
I have made a PR with a suggestion. In short: we needed to decide on who owned each kwarg, and only consume it in one place.